server/src/main/java/org/elasticsearch/index/engine/LiveVersionMapArchiver.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.engine;
+
+import org.apache.lucene.util.BytesRef;
+
+/**
+ * Keeps track of the old map of a LiveVersionMap that gets evacuated on a refresh
+ */
+public interface LiveVersionMapArchiver {
```

Comment:
nit: since this not only archives the data but also allows to get it, I would call this the `LiveVersionMapArchive`.